File: app/src/main/java/com/simpleshift/scheduler/domain/model/RuntimeShiftSettings.kt
package com.simpleshift.scheduler.domain.model

import androidx.compose.runtime.Immutable
import java.time.LocalDate

@Immutable
data class RuntimeShiftSettings(
    val cycleLength: Int = ShiftCycleConfig.CYCLE_LENGTH,
    val shiftCycle: List<ShiftType> = ShiftCycleConfig.SHIFT_CYCLE,
    val defaultTeamId: Int = 1,
    val referenceDate: LocalDate = ShiftCycleConfig.REFERENCE_DATE
) {
    val isValid: Boolean
        get() = cycleLength in 1..100 && shiftCycle.size == cycleLength
                && shiftCycle.all { it in ShiftType.entries }
                && defaultTeamId in 1..Team.TOTAL_TEAMS

    val teamPhaseStep: Int
        get() = cycleLength / Team.TOTAL_TEAMS
}

File: flutter/lib/domain/models/runtime_shift_settings.dart
import 'shift_type.dart';
import 'shift_cycle_config.dart';
import 'team.dart';

/// 运行时周期配置 — 对应 Android 版 RuntimeShiftSettings.kt
class RuntimeShiftSettings {
  final int cycleLength;
  final List<ShiftType> shiftCycle;
  final int defaultTeamId;
  final DateTime referenceDate;

  RuntimeShiftSettings({
    this.cycleLength = ShiftCycleConfig.cycleLength,
    this.shiftCycle = ShiftCycleConfig.shiftCycle,
    this.defaultTeamId = 1,
    DateTime? referenceDate,
  }) : referenceDate = referenceDate ?? ShiftCycleConfig.referenceDate;

  bool get isValid =>
      cycleLength >= 1 &&
      cycleLength <= 100 &&
      shiftCycle.length == cycleLength &&
      shiftCycle.every((t) => ShiftType.values.contains(t)) &&
      defaultTeamId >= 1 &&
      defaultTeamId <= Team.totalTeams;

  int get teamPhaseStep => cycleLength ~/ Team.totalTeams;
}
